tests/lib/c_lib/thrd/src/thrd.c

```diff
@@ -79,7 +79,7 @@ ZTEST(libc_thrd, test_thrd_create_join)
 
 	zassert_equal(thrd_success, thrd_create(&thr, fun, &param));
 	zassert_equal(thrd_success, thrd_join(thr, &res));
-	zassert_equal(BIOS_FOOD, param, "expected: %d actual: %d", BIOS_FOOD, param);
+	zassert_equal(BIOS_FOOD, param, "expected: %lu actual: %lu", BIOS_FOOD, param);
```

Comment:
`BIOS_FOOD` is just a `#define` and doesn't have a trailing 'l', so it'll just be 'int' type. `param` is `uintptr_t`, which is not a type that printf knows about internally. Include `<inttypes.h>` and use `"%" PRIuPTR` for both, adding a cast to `(uintptr_t)` for `BIOS_FOOD` so they match.

Reply:
A safe option would be to cast both parameters to `(unsigned long)` and use "%ul" (this is a error print at the end)

Reply:
yep, a cast + lu would work